aten/src/ATen/native/cuda/Assert.cuh

```diff
@@ -0,0 +1,203 @@
+#pragma once
+
+#include <c10/cuda/CUDAAssert.h>
+#include <c10/macros/Macros.h>
+#include <cstddef>
+
+namespace at {
+namespace native {
+namespace assert {
+
+using namespace c10::cuda;
+
+static inline C10_HOST_DEVICE char* write_string(
```

Comment:
We can delete these now right?

Reply:
This implementation is only used when string-literals would show up in the var-arg list of the macro. If the string support (which is already tested) should be dropped completely the argument-handling for string could be removed. It could also be made optional with a #ifdef so that a specific preprocessor-define would have to be present in order to enable the string processing code.

Reply:
OK. My general preference is that if we leave code around, there should be something exercising it (even a test will do). However, my understanding from your earlier description (though I could be wrong!) was that use of a string literal would cause a hard-to-understand performance regression, so we should make it hard for people to make use of this feature. So maybe an `#ifdef` is the right call here.